pycyphal/transport/udp/_frame.py

```diff
@@ -21,14 +22,21 @@ class UDPFrame(pycyphal.transport.commons.high_overhead_transport.Frame):
     The current header format enables encoding by trivial memory aliasing on any conventional little-endian platform::
 
         struct Header {
-            uint8_t  version;
-            uint8_t  priority;
-            uint16_t _reserved_a;   // Set to zero when encoding, ignore when decoding.
-            uint32_t frame_index_eot;
+            uint4_t  version;               # <- 1
+            uint4_t  _reserved_a;
+            uint3_t  priority;              # Duplicates QoS for ease of access; 0 -- highest, 7 -- lowest.
+            uint5_t  _reserved_b;
+            uint16_t source_node_id;        # 0xFFFF == anonymous transfer
+            uint16_t destination_node_id;   # 0xFFFF == broadcast
+            uint15_t data_specifier;        # subject-ID | (service-ID + RNR (Request, Not Message))
+            bool     snm;                   #SNM (Service, Not Message)
             uint64_t transfer_id;
-            uint64_t _reserved_b;   // Set to zero when encoding, ignore when decoding.
+            uint31_t frame_index;
+            bool     frame_index_eot;
+            uint16_t user_data;             # Opaque application-specific data with user-defined semantics. Generic implementations should ignore
+            uint16_t header_crc;
```

Comment:
There are no non-multiple-of-8-bits standard types so this example does not serve much purpose anymore, I suggest removing it. Either that, or coalescing adjacent non-standard-size fields up to the nearest standard size.

Reply:
Leaving it in for the n00bs